compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -4039,6 +4084,47 @@ abstract class GenJSCode extends plugins.PluginComponent
           js.Undefined()
         else
           genPrimitiveJSClass(classSym)
+      } else if (code == CREATE_INNER_JS_CLASS || code == CREATE_LOCAL_JS_CLASS) {
+        val classSym = resolveReifiedJSClassSym(args(0))
+        val superClassValue = genExpr(args(1))
+        if (classSym == NoSymbol) {
+          js.Undefined()
+        } else {
+          val captureValues = {
+            if (code == CREATE_INNER_JS_CLASS) {
+              val outer = genThis()
+              List.fill(classSym.info.decls.count(_.isClassConstructor))(outer)
+            } else {
+              val ArrayValue(_, fakeNewInstances) = args(2)
+              fakeNewInstances.flatMap(genCaptureValuesFromFakeNewInstance(_))
+            }
+          }
+          js.CreateJSClass(jstpe.ClassRef(encodeClassFullName(classSym)),
+              superClassValue :: captureValues)
+        }
+      } else if (code == NEW_INNER_LOCAL_JS_CLASS) {
+        val jsClassValue = genExpr(args(0))
+        val newTree = args(1)
+        assert(newTree.isInstanceOf[Apply], pos)
+        genPrimitiveJSNew(newTree.asInstanceOf[Apply], Some(jsClassValue))
+      } else if (code == NEW_INNER_LOCAL_JS_OBJECT) {
+        val jsSuperClassValue = genExpr(args(0))
+        val newTree = args(1)
+        assert(newTree.isInstanceOf[Apply], pos)
+        genPrimitiveJSNew(newTree.asInstanceOf[Apply], Some(jsSuperClassValue))
```

Comment:
It's a bit weird. It's because the semantics of the class value are not the same: in the JS_CLASS case, it refers to the class value of that very class, but in the JS_OBJECT, it refers to the class value *of the super class*.

I was also thinking of merging both of them with `INNER_LOCAL_JS_SUPER_CALL`, and simply keep a unique `WITH_CONTEXTUAL_JS_CLASS_VALUE` primitive that would store the class value in `contextualJSCLassValue` (replacing `jsSuperClassValueForSuperCall`).